src/Domain/Item/ItemService.ts

```diff
@@ -0,0 +1,75 @@
+import * as dayjs from 'dayjs'
+import { inject, injectable } from 'inversify'
+import TYPES from '../../Bootstrap/Types'
+import { GetItemsDTO } from './GetItemsDTO'
+
+import { GetItemsResult } from './GetItemsResult'
+import { Item } from './Item'
+import { ItemQuery } from './ItemQuery'
+import { ItemRepositoryInterface } from './ItemRepositoryInterface'
+import { ItemServiceInterface } from './ItemServiceInterface'
+import { SaveItemsResult } from './SaveItemsResult'
+
+@injectable()
+export class ItemService implements ItemServiceInterface {
+  private readonly DEFAULT_ITEMS_LIMIT = 100000
+  private readonly SYNC_TOKEN_VERSION = 2
+
+  constructor (
+    @inject(TYPES.ItemRepository) private itemRepository: ItemRepositoryInterface,
+  ) {
+  }
+
+  async getItems(dto: GetItemsDTO): Promise<GetItemsResult> {
+    let lastSyncTime = this.getLastSyncTime(dto)
+
+    const itemQuery: ItemQuery = {
+      userUuid: dto.userUuid,
+      lastSyncTime,
+      contentType: dto.contentType,
+      deleted: lastSyncTime ? undefined : false,
+      sortBy: 'updatedAt',
+      limit: !dto.limit || dto.limit < 1 ? this.DEFAULT_ITEMS_LIMIT : dto.limit
+    }
+
+    const items = await this.itemRepository.findAll(itemQuery)
+
+    lastSyncTime = items[items.length - 1].updatedAt
```

Comment:
You need to add 1 microsecond to this otherwise the subsequent sync will return this same item again and again. https://github.com/standardnotes/syncing-server/blob/develop/lib/sync_engine/2019_05_20/sync_manager.rb#L19

Reply:
Fixed one line below :)